### ISSUE TITLE
Crashes on print options screen

### ISSUE THREAD

**Issue by wiraki** [sawhney17/logseq-pdf-export#12]
Hi, thanks for the plugin!

Unfortunately, once I click "Download" to save the PDF, the printing options menu comes up and it just gets stuck, and after a while I get the option to force quit, so the whole Logseq crashes (see screenshot below). This is on Pop!_OS 22.04 (Ubuntu-based). Not sure where to start looking to troubleshoot this one.

![image](https://user-images.githubusercontent.com/11855154/165542813-e817d45d-bf72-4b4f-a863-1dbd8e0ea7e3.png)

**Comment by nickpnmrv** [comment]
Same behaviour for me, I've seen a pattern: once I embed a YouTube video on a page, Logseq starts not responding. Attaching an example file
[logseq-pdf-export test page 1.md](https://github.com/sawhney17/logseq-pdf-export/files/8650586/logseq-pdf-export.test.page.1.md)

**Comment by sawhney17** [comment]
Which option are you using?

**Comment by nickpnmrv** [comment]
I use Print with Retained Formatting. Awesome feature! It looks great for other pages.

**Comment by nickpnmrv** [comment]
It looks like any "embed" creates this problem, I have tried also embedding a block from other page, same behaviour

**Comment by L0cutus** [comment]
Same problem here, Linux Mint, default template, when i click on download button, Logseq crash.
Sometimes i can see the printer dialog box before it crash, sometimes it crash immediately.

**Comment by life5ign** [comment]
Same exact issue here; Ubuntu 22.04 as PopOS.  This looks like a great plugin, and there is no native functionality for PDF exports; really hoping to get this working!

**Comment by DvalinT** [comment]
Had the same issue, I managed to "solve it" by disconnecting myself from Wifi: I am tempted to say that this crash is due to the fact that my wifi is connected to my printer (HP something) good luck with that cheers

**Comment by kevdotpub** [comment]
> Had the same issue, I managed to "solve it" by disconnecting myself from Wifi: I am tempted to say that this crash is due to the fact that my wifi is connected to my printer (HP something) good luck with that cheers

This also works for me and I also have a Wifi printer and am also on Ubuntu (22.10). Hope this helps someone!

**Comment by sawhney17** [comment]
Interesting. Glad it's fixed. I'll add a disclaimer in the readme.

**Comment by sawhney17** [comment]
@mpauper the truth is I have no idea whatsoever how to fix this. Looks to me like an upstream electron issue on Linux. Don't have a Linux desktop to test and I'm using the native electron print command. Hands are a bit tied. Marking this post as help wanted. Hopefully the disclaimer helps some people.

**Comment by alxlg** [comment]
@life5ign because you commented with zero additional information, sending an email to everyone who is subscribed to this. It's seen as something very bad on GitHub and other platforms.

**Comment by life5ign** [comment]
@alxlg  lmfao!  you thumbs downed my question as well.  get a grip man. I suggest you get out from behind your screen, look at what's happening in the world, and reserve the use of "very bad" to other things.  This is completely subjective btw; stack overflow open source won't even let me post a question on it, because AI detected that it's subjective.

Also, grouchily and lazily giving a thumbs down doesn't tell the poster any useful information about why you're doing this, and deters them from interacting with the project.  Instead, you could summon the courage to explain yourself.  

Finally, for the most important point:  me indicating that I have the same issue on the same distro and version *isn't zero additional information*   The best you couldn't have done here was suggest that, on this project, some people (you) prefer that a thumbs up on the OP should be given, if you're having the same issue on the same platform, which most people would instantly realize is preference.

Summary: stop being a jerk.  Imagine yourself irl literally doing a thumbs down when someone offered information (yes, the distro) or asked a question. 

By the way, your behavior led to *three* additional emails to the group watching this.  Go ahead, reply to this, and add another! I dare you to violate your own principles.  Oh wait, you have nothing to say, actually... let me guess: thumbs down, lmfao.

Now, I'm going to go forget about this.  Thanks to the helpful people on this thread.

**Comment by jiriks74** [comment]
The crashes I experienced werer cause by the plugin `Banners`. Turning it off allowed me to do a `Print to PDF` successfully with `Retainer formatting`.

If you use `Tabs` or `Awesome Styler` I recommend changing the theme to a default one and turning off `TabsI` as it results in a better pdf.

`Tabs` leaves tabs on top of each pdf page:
![image](https://user-images.githubusercontent.com/54378412/225232464-6fe80459-b8ef-408e-beef-915f9cfecebf.png)

While `Awesome Styler`'s theme leaves a weird border on the first page of the pdf and a bit of the same border on the second page
First page:
![image](https://user-images.githubusercontent.com/54378412/225232769-60649544-807f-411b-94a1-8e971ad11c3f.png)
End of first page, beginning of second page
![image](https://user-images.githubusercontent.com/54378412/225232939-40f68c98-d86b-4479-8b3a-a798455b702d.png)

TLDR: Turn off Banners if you use them.

**Comment by kumaresh0** [comment]
I encounter the same issue with MacOS, plugins other functionalities work great like native. but when embedding video on the page then even with retain formatting option logseq just stuck I have to force quit and relaunch :(
